### DISCUSSION TITLE
Dapper

### DISCUSSION THREAD

**Discussion by etaptecnologia** [balta-io/forum#164] [Tira Dúvidas] [answered]
Boa tarde comunidade Balta

Abaixo existe um código.

Neste código existe uma propriedade public int EnderecoId { get; set; }, eu gostaria de saber se é possível efetuar o insert no meu banco de dados usando dapper sem precisar dessa propriedade setada na minha classe,

Acredito que não fica elegante ter essa propriedade em minha classe, pois a unica utilidade dela é efetuar insert no banco.

Obrigado

`using Dapper.Contrib.Extensions;

namespace ConsoleApp3.Model
{
    [Table("[Instituicao]")]
    internal class Instituicao
    {
        public int Id { get; set; }
        public string Nome { get; set; }

        public int EnderecoId { get; set; }

        [Write(false)]
        public Endereco Endereco { get; set; }
    }
}
`

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @etaptecnologia, como vai?
A propriedade `Endereco` é umm propriedade de apoio, você poderá preencher ou não os dados do endereço quando carregar uma instituição. Ela existe apenas para trazer esta facilidade.

Se você não vai usar ela, pode remover sim. Já a propriedade `EnderecoId` é a relação entre as tabelas e deve ser mantida.

**Reply by etaptecnologia** [reply]
Perfeito, muito obrigado!
